jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +66,88 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the file.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath) throws IOException {
+    TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider();
+    Path tempDirectoryPath = tempDirectoryProvider.newDirectory();
+    ZipUtil.unzip(jarPath, tempDirectoryPath);
+
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isResourceFile = path -> !path.getFileName().toString().endsWith(".class");
+
+    // Determine class and resource files in the directory containing jar contents and create
+    // FileEntriesLayer for each type of layer (class or resource), while maintaining the
+    // file's original project structure.
+    FileEntriesLayer classesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                tempDirectoryPath,
+                isClassFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Classes")
+            .build();
+    FileEntriesLayer resourcesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                tempDirectoryPath,
+                isResourceFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Resources")
+            .build();
+
+    // Get dependencies from Class-Path in the jar's manifest and create a FileEntriesLayer.Builder
+    // with these dependencies as entries.
+    List<Path> dependencies = new ArrayList<>();
+    JarFile jarFile = new JarFile(jarPath.toFile());
+    String classPath = jarFile.getManifest().getMainAttributes().getValue("Class-Path");
+    if (classPath != null) {
+      for (String dependency : Splitter.on(" ").split(classPath)) {
```

Comment:
`" "` -> `"//s+"` for handling any kind of white space
I might also do `classPath.trim()` ?

Reply:
May be a good idea. For example, [here](https://stackoverflow.com/questions/3057841/too-long-line-in-manifest-file-while-trying-to-create-jar), I see people saying like

> two spaces immediately following `Class-Path:`, a space at the end of [each] line, and one before each line, and finally an extra new-line at the end did the trick.

although I'm not sure the Java API handles all these whitespaces correctly. Worth checking anyway.

Reply:
Oh great point, added handling for any kind of white space and added extra spaces in the class-path field in the sample jar file  to make sure this is working.